src/redisearch_rs/encode_decode/src/varint/vector_writer.rs

```diff
@@ -0,0 +1,73 @@
+use super::write;
+
+#[derive(Debug)]
+pub(crate) struct VectorWriter {
+    buffer: Vec<u8>,
+    n_members: usize,
+    last_value: u32,
+}
+
+impl VectorWriter {
+    /// Create a new `VectorWriter` with the given capacity.
+    pub fn new(cap: usize) -> Self {
+        Self {
+            buffer: Vec::with_capacity(cap),
+            last_value: 0,
+            n_members: 0,
+        }
+    }
+
+    /// Write an integer to the vector.
+    ///
+    /// # Return Value
+    ///
+    /// The number of bytes written to the vector.
+    pub fn write(&mut self, value: u32) -> std::io::Result<usize> {
+        // Assumes `value` is greater than `last_value` (copied over from the C code).
+        let bytes_written = write(value - self.last_value, &mut self.buffer)?;
```

Comment:
Worth calling this out in the function documentation.
Should we include a `debug_assert` here to ensure the invariant isn't violated in tests? Unless we have tests exercising this specific scenario already.